lib/portage/sync/modules/rsync/rsync.py

```diff
@@ -230,15 +230,19 @@ def update(self):
 			addrinfos = None
 			uris = []
 
-			try:
-				addrinfos = getaddrinfo_validate(
-					socket.getaddrinfo(getaddrinfo_host, None,
-					family, socket.SOCK_STREAM))
-			except socket.error as e:
-				writemsg_level(
-					"!!! getaddrinfo failed for '%s': %s\n"
-					% (_unicode_decode(hostname), str(e)),
-					noiselevel=-1, level=logging.ERROR)
+			rsync_proxy_conf = \
+				self.options['emerge_config'].running_config.settings.get(
+					'RSYNC_PROXY', None)
```

Comment:
The environment used to spawn rsync is `self.spawn_kwargs['env']`, so that's where the real RSYNC_PROXY value will come from.

Reply:
From my humble comprehension and tests I've made, here the issue for getting addr info is that your socket does not handle proxy settings. I've made a version handling proxy setting for `socket` but it's rather hacky. I use `socks.setdefaultproxy` and passing `socks.socksocket` to `socket.socket`. Then, I've made it working. But finally, failing on `getaddrinfo` is not failing whole `rsync`. Else, I'm missing some feature of `socket` library handling system-wide proxy.